src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java

```diff
@@ -0,0 +1,37 @@
+package com.znsio.sample.e2e.businessLayer.amazon;
+
+import com.context.TestExecutionContext;
+import com.znsio.e2e.entities.Platform;
+import com.znsio.e2e.runner.Runner;
+import com.znsio.sample.e2e.entities.SAMPLE_TEST_CONTEXT;
+import com.znsio.sample.e2e.screen.amazon.AmazonHomeScreen;
+import org.apache.log4j.Logger;
+
+public class AmazonHomePageBL {
+    private static final Logger LOGGER = Logger.getLogger(AmazonHomePageBL.class.getName());
+    private final TestExecutionContext context;
+    private final String currentUserPersona;
+    private final Platform currentPlatform;
+
+    public AmazonHomePageBL(String userPersona, Platform forPlatform){
+        long threadId = Thread.currentThread()
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        this.currentUserPersona = userPersona;
+        this.currentPlatform = forPlatform;
+        Runner.setCurrentDriverForUser(userPersona, forPlatform, context);
+    }
+
+    public AmazonHomePageBL() {
+        long threadId = Thread.currentThread()
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+    }
+
+    public SearchResultListPageBL searchForIphone13() {
+        AmazonHomeScreen.get().searchIphone13();
+        return new SearchResultListPageBL();
```

Comment:
no need to create object of same class

Reply:
@vishal07111998 the above function is created in AmazonHomePageBL class and it's return SearchResultListPageBL class object.